pkg/kubelet/dockertools/manager.go

```diff
@@ -1280,6 +1363,11 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 	containersToKeep := make(map[kubeletTypes.DockerID]int)
 	createPodInfraContainer := false
 
+	if pod.DeletionTimestamp != nil {
```

Comment:
Just a thought: If we move this logic to [syncPod in kubelet.go](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1081), and just call killPod from there, we can avoid having to inject this logic in each container runtime.